tools/mafft/mafft.xml

```diff
@@ -1,229 +1,289 @@
-    
-    - mafft --globalpair --maxiterate 1000 input [> output]
-
-- E-INS-i (suitable for sequences containing large unalignable regions; recommended for <200 sequences):
-    
-    - mafft --ep 0 --genafpair --maxiterate 1000 input [> output]. For E-INS-i, the --ep 0 option is recommended to allow large gaps. 
-
-
-**Speed-oriented methods:**
-
-- FFT-NS-i (iterative refinement method; two cycles only):
-
-    - mafft --retree 2 --maxiterate 2 input [> output]
-
-- FFT-NS-i (iterative refinement method; max. 1000 iterations):
-
-    - mafft --retree 2 --maxiterate 1000 input [> output]
-
-- FFT-NS-2 (fast; progressive method):
-
-    - mafft --retree 2 --maxiterate 0 input [> output]
-
-- FFT-NS-1 (very fast; recommended for >2000 sequences; progressive method with a rough guide tree):
-
-    - mafft --retree 1 --maxiterate 0 input [> output]
-
-- NW-NS-i (iterative refinement method without FFT approximation; two cycles only):
-
-    - mafft --retree 2 --maxiterate 2 --nofft input [> output]
-
-- NW-NS-2 (fast; progressive method without the FFT approximation):
-
-    - mafft --retree 2 --maxiterate 0 --nofft input [> output]
-
-- NW-NS-PartTree-1 (recommended for ~10,000 to ~50,000 sequences; progressive method with the PartTree algorithm):
-
-    - mafft --retree 1 --maxiterate 0 --nofft --parttree input [> output]
-
-**Options:**
-
---auto
-    Automatically selects an appropriate strategy from L-INS-i, FFT-NS-i and FFT-NS-2, according to data size. Default: off (always FFT-NS-2)
---adjustdirection
-    Generate reverse complement sequences, as necessary, and align them together with the remaining sequences. In the case of protein alignment, these options are just ignored.
---op
-    Gap opening penalty, default: 1.53
---ep
-    Offset (works like gap extension penalty), default: 0.0
---maxiterate
-    Maximum number of iterative refinement, default: 0
---clustalout
-    Output: clustal format, default: fasta
---thread
-    Number of threads (if unsure, --thread -1)
---retree number
-    Guide tree is built number times in the progressive stage.  Valid with 6mer distance.  Default: 2
-    ]]>
-    </help>
-    <citations>
-        <citation type="doi">10.1093/molbev/mst010</citation>
-    </citations>
-</tool>
+<?xml version="1.0" encoding="UTF-8"?>
+<tool id="rbc_mafft" name="MAFFT" version="@TOOL_VERSION@+galaxy@VERSION@">
+  <description>Multiple alignment program for amino acid or nucleotide sequences</description>
+  <macros>
+    <import>macros.xml</import>
+  </macros>
+  <expand macro="requirements" />
+  <stdio>
+    <exit_code range="1:" level="fatal" description="Error occurred. Please check Tool Standard Error" />
+    <exit_code range=":-1" level="fatal" description="Error occurred. Please check Tool Standard Error" />
+  </stdio>
+  <version_command>    <![CDATA[
+    mafft --version
+    ]]>
+  </version_command>
+  <command>
+    <![CDATA[
+      export FASTA_4_MAFFT=`which fasta36`;

```

Comment:
Why is this needed?

Reply:
Mafft ask for the fasta3x path to be exported in FASTA_4_MAFFT to use the --fastapair option. I also could export it only if the --fastapair option is selected..